pkg/apis/networking/types.go

```diff
@@ -55,6 +55,18 @@ type NetworkPolicySpec struct {
 	// solely to ensure that the pods it selects are isolated by default)
 	// +optional
 	Ingress []NetworkPolicyIngressRule
+
+	// List of egress rules to be applied to the selected pods. Outgoing traffic is
+	// allowed if there are no NetworkPolicies selecting the pod (and cluster policy
+	// otherwise allows the traffic), OR if the traffic matches at least one egress rule
+	// across all of the NetworkPolicy objects whose podSelector matches the pod. If
+	// this field is empty then this NetworkPolicy does not limit any outgoing traffic
+	// (and serves solely to ensure that the pods it selects does not allow any outgoing traffic.)
+	// +optional
+	Egress []NetworkPolicyEgressRule
+
+	// +optional
+	PolicyType []string
```

Comment:
I think most other types are defined like this:

```go
type PolicyType string

const (
    PolicyTypeIngress PolicyType = "Ingress"
)
```

For reference: https://github.com/kubernetes/kubernetes/blob/master/pkg/api/types.go#L2693

Reply:
I think this should be `PolicyTypes` (plural)

Reply:
@danwinship I think you were against this approach but I can't find the comment.  Any reasons? This seems consistent with other similar API fields.

Reply:
No, that was on an earlier version of the IPBlock PR where he was using a string alias type for CIDR values. I agree we should use an alias type for "enum-like" values.